packages/core/test/combinator/slice/bounds-test.js

```diff
@@ -0,0 +1,75 @@
+import { assert } from '@briancavalier/assert'
+import { describe, it } from 'mocha'
+
+import {
+  isInfiniteBounds,
+  isNilBounds,
+  maxBounds,
+  mergeBounds,
+  minBounds,
+  newBounds
+} from '../../../src/combinator/slice/bounds'
+
+const rInt = (min = 0, max = Number.MAX_SAFE_INTEGER) =>
+  min + (Math.random() * (max - min))
+
+const genBound = () =>
+  Math.random() >= 0 ? rInt() : -rInt()
+
+const assertValidBounds = gen => {
+  for (let i = 0; i < 100; ++i) {
+    const b = gen(i)
+    assert(b.min >= 0 && b.max >= b.min)
+  }
+}
+
+describe('slice/bounds', () => {
+  describe('newBounds', () => {
+    it('should create valid bounds', () => {
+      assertValidBounds(() => newBounds(genBound(), genBound()))
```

Comment:
There is a small possibility that this test could randomly fail and pass. If the subject under test’s source is modified actually to be a wrong implementation, the test could still pass should the 100 iterations result in a non-gaussian distribution.

Reply:
Afaik, Math.random does generate a normal (gaussian) distribution.  So, this is effectively property testing over a normal distro.  We could up the iterations to 500 or 1000, or add a few exact tests specifically to test at the boundary conditions ... or both.

Reply:
I have made a few tests, and it seems that 100 iterations produce a reasonably normal distribution most of the time.